pkg/assisted/agentserviceconfig_test.go

```diff
@@ -0,0 +1,810 @@
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestAgentServiceConfigBuilder()
+		testBuilder.WithMirrorRegistryRef(testCase.mirrorRegistryConfig)
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.mirrorRegistryConfig, testBuilder.Definition.Spec.MirrorRegistryRef.Name)
+		}
+	}
+}
+
+//nolint:lll
+func TestAgentServiceConfigWithOSImage(t *testing.T) {
+	testCases := []struct {
+		osImages      []agentInstallV1Beta1.OSImage
+		expectedError string
+	}{
+		{
+			osImages: []agentInstallV1Beta1.OSImage{
+				{
+					OpenshiftVersion: "4.12",
+					Version:          "412.86.202402272018-0",
+					Url:              "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.12/4.12.30/rhcos-live.x86_64.iso",
+					CPUArchitecture:  "x86_64",
+				},
+			},
+			expectedError: "",
+		},
+		{
+			osImages: []agentInstallV1Beta1.OSImage{
+				{
+					OpenshiftVersion: "4.12",
+					Version:          "412.86.202402272018-0",
+					Url:              "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.12/4.12.30/rhcos-live.x86_64.iso",
+					CPUArchitecture:  "x86_64",
+				},
+				{
+					OpenshiftVersion: "4.13",
+					Version:          "413.92.202307260246-0",
+					Url:              "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.13/4.13.10/rhcos-live.x86_64.iso",
+					CPUArchitecture:  "x86_64",
+				},
+			},
+			expectedError: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestAgentServiceConfigBuilder()
+
+		for _, image := range testCase.osImages {
+			testBuilder.WithOSImage(image)
+		}
+
+		assert.Equal(t, testCase.osImages, testBuilder.Definition.Spec.OSImages)
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+	}
+}
+func TestAgentServiceConfigWithUnauthenticatedRegistry(t *testing.T) {
+	testCases := []struct {
+		resgistries   []string
+		expectedError string
+	}{
+		{
+			resgistries: []string{
+				"docker.io",
+			},
+			expectedError: "",
+		},
+		{
+			resgistries: []string{
+				"docker.io",
+				"quay.io",
+			},
+			expectedError: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestAgentServiceConfigBuilder()
+
+		for _, registry := range testCase.resgistries {
+			testBuilder.WithUnauthenticatedRegistry(registry)
+		}
+
+		assert.Equal(t, testCase.resgistries, testBuilder.Definition.Spec.UnauthenticatedRegistries)
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+	}
+}
+
+func TestAgentServiceConfigWithIPXEHTTPRoute(t *testing.T) {
+	testCases := []struct {
+		route         string
+		expectedError string
+	}{
+		{
+			route:         "enabled",
+			expectedError: "",
+		},
+		{
+			route:         "disabled",
+			expectedError: "",
+		},
+		{
+			route:         "",
```

Comment:
We have to rely on our validation instead. We agreed if we want to build invalid object we have to mutate element manually in test case:
// Test invalid Route configuration
builder.Definition.Spec.Route=""

Reply:
I'm not sure I understand. Isn't this test checking my validation?